GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Exceptions.Modify.cs

```diff
@@ -0,0 +1,262 @@
+        {
+            // given
+            int minutesInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+
+            Source randomSource =
+                CreateRandomSource(randomDateTimeOffset);
+
+            randomSource.CreatedDate =
+                randomDateTimeOffset.AddMinutes(minutesInPast);
+
+            var dbUpdateException = new DbUpdateException();
+
+            var failedOperationSourceException =
+                new FailedOperationSourceException(
+                    message: "Failed operation source  error occurred, contact support.",
+                    innerException: dbUpdateException);
+
+            var expectedSourceDependencyException =
+                new SourceDependencyException(
+                    message: "Source dependency error occurred, contact support.",
+                    innerException: failedOperationSourceException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectSourceByIdAsync(randomSource.Id))
+                    .ThrowsAsync(dbUpdateException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(randomDateTimeOffset);
+
+            // when
+            ValueTask<Source> modifySourceTask =
+                this.sourceService.ModifySourceAsync(randomSource);
+
+            SourceDependencyException actualSourceDependencyException =
+                await Assert.ThrowsAsync<SourceDependencyException>(
+                    modifySourceTask.AsTask);
+
+            // then
+            actualSourceDependencyException.Should().BeEquivalentTo(
+                expectedSourceDependencyException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectSourceByIdAsync(randomSource.Id),
+                    Times.Once);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogErrorAsync(It.Is(SameExceptionAs(
+                    expectedSourceDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        private async Task ShouldThrowDependencyValidationExceptionOnModifyIfDbUpdateConcurrencyOccursAndLogItAsync()
+        {
+            // given
+            int minutesInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+
+            Source randomSource =
+                CreateRandomModifySource(randomDateTimeOffset);
+
+            randomSource.CreatedDate = randomDateTimeOffset.AddMinutes(minutesInPast);
```

Comment:
We do not have to do the date in past logic as it is not needed for this test.  
We can just construct a valid model since we will throw the exception on the first dependency and as that is the dateTimeBroker
we will not reach the date validation logic.

```suggestion
            Source randomSource = CreateRandomSource(randomDateTimeOffset);
```